extmod/modubinascii.c

```diff
@@ -58,10 +62,43 @@ STATIC mp_obj_t mod_binascii_hexlify(mp_uint_t n_args, const mp_obj_t *args) {
 }
 MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(mod_binascii_hexlify_obj, 1, 2, mod_binascii_hexlify);
 
+STATIC mp_obj_t mod_binascii_unhexlify(mp_obj_t data) {
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(data, &bufinfo, MP_BUFFER_READ);
+
+    if ((bufinfo.len & 1) != 0) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ubinascii_dot_Error, "Odd-length string"));
```

Comment:
Yeah - I wasn't sure how close to binascii we should be. I'm happy either way. I figured I'd throw this out and see what responses I got :)